src/modules/jcms_rest/src/Plugin/rest/resource/HighlightListRestResource.php

```diff
@@ -0,0 +1,119 @@
+ *   uri_paths = {
+ *     "canonical" = "/highlights/{list}"
+ *   }
+ * )
+ */
+class HighlightListRestResource extends AbstractRestResourceBase {
+
+  /**
+   * Responds to GET requests.
+   *
+   * Returns a list of bundles for specified entity.
+   *
+   * @param string $list
+   * @return array|\Symfony\Component\HttpFoundation\JsonResponse
+   *
+   * @throws \Symfony\Component\HttpKernel\Exception\HttpException
+   *   Throws exception expected.
+   */
+  public function get($list) {
+    $query = \Drupal::entityQuery('node')
+      ->condition('status', NODE_PUBLISHED)
+      ->condition('changed', REQUEST_TIME, '<')
+      ->condition('type', 'highlight_list')
+      ->condition('title', $list);
+    $response = [];
+
+    $nids = $query->execute();
+    if ($nids) {
+      $nid = reset($nids);
+      /* @var \Drupal\node\Entity\Node $node */
+      $node = \Drupal\node\Entity\Node::load($nid);
+
+      if ($node->get('field_highlight_items')->count()) {
+        foreach ($node->get('field_highlight_items')->referencedEntities() as $item) {
+          $response[] = $this->getHighlightItem($item);
+        }
+      }
+    }
+    else {
+      $query = Database::getConnection()->select('node__field_highlight_item', 'hi');
+      $query->addField('hi', 'entity_id', 'item');
+      $query->condition('hi.bundle', 'highlight_item');
+      $query->innerJoin('node__field_highlight_items', 'his', 'his.field_highlight_items_target_id = hi.entity_id');
+      $query->addField('his', 'delta');
+      $query->orderBy('his.delta', 'ASC');
+      $query->leftJoin('node__field_subjects', 's', 's.entity_id = hi.field_highlight_item_target_id');
+      $query->leftJoin('taxonomy_term__field_subject_id', 'si', 'si.entity_id = s.field_subjects_target_id');
+      $query->addField('si', 'field_subject_id_value', 'subject_id');
+      $query->leftJoin('node__field_episode_chapter', 'ec', 'ec.field_episode_chapter_target_id = hi.field_highlight_item_target_id');
+      $query->leftJoin('node__field_subjects', 'sp', 'sp.entity_id = ec.entity_id');
+      $query->leftJoin('taxonomy_term__field_subject_id', 'spi', 'spi.entity_id = sp.field_subjects_target_id');
+      $query->addField('spi', 'field_subject_id_value', 'podcast_subject_id');
+      $db_or = new Condition('OR');
+      $db_or->condition('si.field_subject_id_value', $list);
+      $db_or->condition('spi.field_subject_id_value', $list);
+      $query->condition($db_or);
+
+      if ($results = $query->execute()->fetchAllKeyed()) {
+        /* @var \Drupal\node\Entity\Node[] $items */
+        if ($items = \Drupal\node\Entity\Node::loadMultiple(array_keys($results))) {
+          $response = [];
+          foreach ($items as $item) {
+            $response[] = $this->getHighlightItem($item);
+          }
+        }
+      }
+    }
+
+    if (!empty($response)) {
+      $response = new JsonResponse($response, Response::HTTP_OK, ['Content-Type' => 'application/vnd.elife.highlights+json;version=1']);
```

Comment:
Not for this PR, but we should probably have some sort of static variable/constant for when referencing these content types.